engine/src/main/java/com/datatorrent/stram/plan/physical/PTOperator.java

```diff
@@ -85,6 +85,7 @@
     private static final long serialVersionUID = 201312112033L;
 
     public final LogicalPlan.StreamMeta logicalStream;
+    public Locality locality;
```

Comment:
You need the streammeta between unifier and downstream operator not unifier and upstream operator..